sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/ApplicationGatewayTests.java

```diff
@@ -135,6 +133,72 @@ public void canCRUDApplicationGatewayWithWAF() throws Exception {
                 "REQUEST-943-APPLICATION-ATTACK-SESSION-FIXATION");
     }
 
+    @Test
+    public void canSpecifyWildcardListeners() {
+        String appGatewayName = generateRandomResourceName("agwaf", 15);
+        String appPublicIp = generateRandomResourceName("pip", 15);
+
+        PublicIpAddress pip =
+            networkManager
+                .publicIpAddresses()
+                .define(appPublicIp)
+                .withRegion(Region.US_EAST)
+                .withNewResourceGroup(rgName)
+                .withSku(PublicIPSkuType.STANDARD)
+                .withStaticIP()
+                .create();
+        String listenerName = "listener1";
+        // regular hostname
+        String hostname1 = "my.contoso.com";
+        ApplicationGateway gateway = networkManager.applicationGateways()
+            .define(appGatewayName)
+            .withRegion(Region.US_EAST)
+            .withExistingResourceGroup(rgName)
+
+            // Request routing rules
+            .defineRequestRoutingRule("rule80")
+            .fromPublicFrontend()
+            .fromFrontendHttpPort(80)
+            .toBackendHttpPort(8080)
+            .toBackendIPAddress("11.1.1.1")
+            .toBackendIPAddress("11.1.1.2")
+            .withCookieBasedAffinity()
+            .attach()
+
+            // Additional/explicit frontend listeners
+            .defineListener(listenerName)
+            .withPublicFrontend()
+            .withFrontendPort(9000)
+            .withHttp()
+            .withHostname(hostname1)
+            .attach()
+            .withTier(ApplicationGatewayTier.WAF_V2)
+            .withSize(ApplicationGatewaySkuName.WAF_V2)
+            .withAutoScale(2, 5)
+            .withExistingPublicIpAddress(pip)
+            .create();
+
+        Assertions.assertEquals(hostname1, gateway.listeners().get(listenerName).hostname());
+
+        // wildcard hostname
+        String hostname2 = "*.contoso.com";
+        gateway.update()
+            .updateListener(listenerName)
+            .withHostname(hostname2)
+            .parent()
+            .apply();
+
+        Assertions.assertEquals(hostname2, gateway.listeners().get(listenerName).hostname());
+
+        // override hostnames by hostname
+        gateway.innerModel().httpListeners().iterator().next().withHostNames(null).withHostname(hostname1);
+        gateway.update()
+            .apply();
+
+        Assertions.assertEquals(hostname1, gateway.listeners().get(listenerName).hostname());
```

Comment:
Do we need this test?

Reply:
It's for confirmation for service response of replacing `hostnames` with `hostname`. I wanted to try the other way around but it's much more complicated..

Reply:
You should be able to replace this case with `withHostnames` if you support it?